### PR TITLE
add chaos brush

### PR DESCRIPTION
**Pull request by hayesj1**
Part 2 of a redo of https://github.com/TeamOpenIndustry/ImmersiveRailroading/pull/1096, providing the livery randomizer to help players make better looking long consists, particularly for freight.

### CODE REVIEW

File: src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java [hunk 1]
```diff
@@ -27,4 +41,81 @@ public List<CreativeTab> getCreativeTabs() {
 		return Collections.singletonList(ItemTabs.MAIN_TAB);
 	}
 
+	@Override
+	public List<String> getTooltip(ItemStack stack) {
+		PaintBrushMode pbm = new Data(stack).mode;
+		if (pbm == null) {
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java:47]: AFAICT this is an impossible condition

File: src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java [hunk 2]
```diff
@@ -27,4 +41,81 @@ public List<CreativeTab> getCreativeTabs() {
...
+
+	@Override
+	public void onClickAir(Player player, World world, Player.Hand hand) {
+		if (player.isCrouching()) {
+			if (world.isServer) {
+				ItemStack item = player.getHeldItem(hand);
+				Data data = new Data(item);
+				data.mode = PaintBrushMode.values()[(data.mode.ordinal() + 1) % (PaintBrushMode.values().length)];
+				data.write();
+				player.sendMessage(PlayerMessage.direct("Set mode to: " + data.mode));
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java:64]: We should probably translate this

File: src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java [hunk 3]
```diff
@@ -27,4 +41,81 @@ public List<CreativeTab> getCreativeTabs() {
...
+		int newIdx;
+		switch (data.mode) {
+			case SEQUENTIAL:
+				newIdx = selectNextTextureIndex(texNames.size(), curIdx, player);
+				break;
+			case RANDOM:
+				newIdx = selectRandomTextureIndex(texNames.size(), curIdx);
+				break;
+			default:
+				ImmersiveRailroading.error("Programmer error: invalid PaintBrush mode: " + data.mode + " is not supported");
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java:82]: runtime exception?

File: src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java (hunk 1 above)
Comment (hayesj1) [reply src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java:47]: I had copiedthat from I think ItemManual. will remove.

File: src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java (hunk 3 above)
Comment (hayesj1) [reply src/main/java/cam72cam/immersiverailroading/items/ItemPaintBrush.java:82]: good catch, will replace accordingly.

File: src/main/java/cam72cam/immersiverailroading/library/PaintBrushMode.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package cam72cam.immersiverailroading.library;
+
+public enum PaintBrushMode {
+	SEQUENTIAL,
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/library/PaintBrushMode.java:4]: Do we want to translate these as well?  I'm not super consistent about it through the rest of the code, but it probably makes sense here

Comment (cam72cam) [reply src/main/java/cam72cam/immersiverailroading/library/PaintBrushMode.java:4]: I think you can, see how the coupler enum works

Comment (hayesj1) [reply src/main/java/cam72cam/immersiverailroading/library/PaintBrushMode.java:4]: I see!

File: src/main/java/cam72cam/immersiverailroading/library/ChatText.java [hunk 5]
```diff
@@ -29,6 +29,9 @@ public enum ChatText {
 	RADIO_NOLINK("radio.nolink"),
 	RADIO_CANT_LINK("radio.cant_link"),
 	BRUSH_NO_VARIANTS("brush.no.variants"),
+	BRUSH_NEXT("brush.next"),
+	BRUSH_MODE_SET("brush.mode.set"),
+	BRUSH_MODE_NAME_PREFIX("brush.mode."),
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/library/ChatText.java:34]: This should be a constant string in the other enum.  This one assumes it's a fully qualified translation.

**Pull request merged**